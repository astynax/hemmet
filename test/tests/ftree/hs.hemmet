|hs|src/{*lib/util data/text/extra} test/{lib-spec data/text/extra-spec} README.md !test-cases/!test-case
