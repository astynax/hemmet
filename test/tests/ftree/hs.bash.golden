#!/bin/bash
cat <<PREVIEW_END
This file tree will be created:
.
├── README.md
├── Src/
│   ├── Data/
│   │   └── Text/
│   │       └── Extra.hs
│   ├── Lib/
│   │   └── Util.hs
│   └── Lib.hs
├── Test/
│   ├── Data/
│   │   └── Text/
│   │       └── ExtraSpec.hs
│   └── LibSpec.hs
└── test-cases/
    └── test-case
PREVIEW_END
read -p "Press any key to continue..." -n1 -s
set -euf -o pipefail
touch "README.md"
mkdir "Src" && pushd "Src"
  mkdir "Data" && pushd "Data"
    mkdir "Text" && pushd "Text"
      touch "Extra.hs"
    popd
  popd
  mkdir "Lib" && pushd "Lib"
    touch "Util.hs"
  popd
  touch "Lib.hs"
popd
mkdir "Test" && pushd "Test"
  mkdir "Data" && pushd "Data"
    mkdir "Text" && pushd "Text"
      touch "ExtraSpec.hs"
    popd
  popd
  touch "LibSpec.hs"
popd
mkdir "test-cases" && pushd "test-cases"
  touch "test-case"
popd
